File: src/main/java/com/github/alexandrecarlton/idea/settings/layout/languages_frameworks/javascript/JavascriptSettings.java
package com.github.alexandrecarlton.idea.settings.layout.languages_frameworks.javascript;

import com.fasterxml.jackson.databind.annotation.JsonDeserialize;
import org.immutables.value.Value;

import java.util.Optional;

@Value.Immutable
@JsonDeserialize(as = ImmutableJavascriptSettings.class)
public interface JavascriptSettings {

  Optional<JavascriptLanguageVersion> javascriptLanguageVersion();

}

File: src/main/kotlin/com/github/alexandrecarlton/idea/settings/layout/languages_frameworks/javascript/JavascriptSettings.kt
package com.github.alexandrecarlton.idea.settings.layout.languages_frameworks.javascript

import com.fasterxml.jackson.annotation.JsonProperty
import com.github.alexandrecarlton.idea.settings.layout.languages_frameworks.javascript.code_quality_tools.JavascriptCodeQualityToolsSettings

data class JavascriptSettings(
    @JsonProperty("JavaScript language version")
    val javascriptLanguageVersion: JavascriptLanguageVersion? = null,

    @JsonProperty("Code Quality Tools")
    val codeQualityTools: JavascriptCodeQualityToolsSettings? = null
)
